includes/Modules/Analytics.php

```diff
@@ -962,6 +962,28 @@ function ( $dimension_def ) {
 					}
 				}
 
+				$dimension_filters          = $data['dimensionFilters'];
+				$dimension_filter_instances = array();
+				if ( ! empty( $dimension_filters ) && is_array( $dimension_filters ) ) {
+					foreach ( $request_args['dimensions'] as $dimension ) {
```

Comment:
This is not correct. `dimensionFilters` and `dimensions` should be completely decoupled, the arguments are not related, unless I'm missing something. To validate/normalize the `dimensionFilters` argument, you should only need to look at `dimensionFilters`, not `dimensions`.

* `dimensionFilters` ensures only results that satisfy a certain dimension criteria are included in the response.
* `dimensions` modifies the response format to group results by certain dimensions.

Reply:
Updated.